plugins/azure/fix_plugin_azure/resource/compute.py

```diff
@@ -557,6 +561,461 @@ class AzureDiskSecurityProfile:
+
+    def post_process(self, graph_builder: GraphBuilder, source: Json) -> None:
+        # Set common storage types
+        if (volume_type := self.volume_type) and (volume_type not in ["UltraSSD_LRS", "PremiumV2_LRS"]):
+            if not (information_name := self.information_name) or not (tier := self.tier):
+                return
+            disk_configuration = storage_sku_info.get(information_name)
+            if not disk_configuration:
+                return
+
+            disk_info = disk_configuration.get(tier)
+            if not disk_info:
+                return
+
+            self.volume_size = disk_info.get("size")
+            self.volume_iops = disk_info.get("maxIOPS")
+            self.volume_throughput = disk_info.get("maxThroughput")
+
+    @staticmethod
+    def _get_ultra_and_prem_v2_sku(location: str) -> List[Json]:
+        ultra_ssd_list = []
+        for disk_size, attributes in ultra_disk_sku_info.items():
+            # Create empty ultra disk object
+            ulta_ssd_object = {
+                "size": disk_size,
+                "skuName": "Ultra SSD",
+                "type": "UltraSSD_LRS",
+                "armRegionName": location,
+                **attributes,
+            }
+            ultra_ssd_list.append(ulta_ssd_object)
+        # Create empty premium ssd v2 object
+        premium_ssd_v2_object = {
+            "skuName": "Premium SSD V2",
+            "type": "PremiumV2_LRS",
+            "armRegionName": location,
+        }
+        return [premium_ssd_v2_object] + ultra_ssd_list
+
+    @classmethod
+    def collect_resources(
+        cls: Type[AzureResourceType], builder: GraphBuilder, **kwargs: Any
+    ) -> List[AzureResourceType]:
+        log.debug(f"[Azure:{builder.subscription.id}] Collecting {cls.__name__} with ({kwargs})")
+        product_names = {"Standard SSD Managed Disks", "Premium SSD Managed Disks", "Standard HDD Managed Disks"}
+        sku_items = []
+        for product_name in product_names:
+            api_spec = AzureApiSpec(
+                service="compute",
+                version="2023-01-01-preview",
+                path=f"https://prices.azure.com/api/retail/prices?$filter=productName eq '{product_name}' and armRegionName eq "
+                + "'{location}' and unitOfMeasure eq '1/Month' and serviceFamily eq 'Storage' and type eq 'Consumption' and isPrimaryMeterRegion eq true",
+                path_parameters=["location"],
+                query_parameters=["api-version"],
+                access_path="Items",
+                expect_array=True,
+            )
+
+            items = builder.client.list(api_spec, **kwargs)
+            sku_items.extend(items)
+        if (location := builder.location) and (location_name := location.name):
+            # Add sizes which we can't fetch from Azure prices API
+            ultra_and_premium_v2_skus = AzureDiskType._get_ultra_and_prem_v2_sku(location_name)
+            sku_items.extend(ultra_and_premium_v2_skus)
```

Comment:
Instead of creating "empty" disk types here - you can move the whole logic to `after_collect` by walking the list of disks.